bciers/apps/reporting/src/app/components/activities/ActivityForm.tsx

```diff
@@ -0,0 +1,200 @@
+"use client";
+import FormBase from "@bciers/components/form/FormBase";
+import { customizeValidator } from "@rjsf/validator-ajv8";
+import { useEffect, useState } from "react";
+import { actionHandler } from "@bciers/actions";
+import { Alert, Button } from "@mui/material";
+import ReportingTaskList from "@bciers/components/navigation/reportingTaskList/ReportingTaskList";
+import { TaskListElement } from "@bciers/components/navigation/reportingTaskList/types";
+
+const tasklistData: TaskListElement[] = [
+  {
+    type: "Page",
+    title: "main page element",
+  },
+  {
+    type: "Section",
+    title: "Facility 1 info",
+    elements: [
+      { type: "Page", title: "Review information", isChecked: true },
+      {
+        type: "Subsection",
+        title: "Activities information",
+        isExpanded: true,
+        elements: [
+          {
+            type: "Page",
+            title: "General stationary combustion",
+          },
+          { type: "Page", title: "Mobile combustion", isActive: true },
+          { type: "Page", title: "Other 1..", isChecked: true },
+        ],
+      },
+      { type: "Page", title: "Non-attributable emissions" },
+    ],
+  },
+  {
+    type: "Section",
+    title: "Facility 2 info",
+    elements: [
+      { type: "Page", title: "Review2 ..." },
+      { type: "Page", title: "Other 2.." },
+    ],
+  },
+  {
+    type: "Section",
+    title: "Facility 3 info",
+    isChecked: true,
+    elements: [
+      { type: "Page", title: "Review3 ..." },
+      { type: "Page", title: "Other 3.." },
+    ],
+  },
+  { type: "Page", title: "New entrant information", isChecked: true },
+  { type: "Page", title: "Operation emission summary with a long title" },
+];
+
+interface Props {
+  activityData: {
+    activityId: number;
+    sourceTypeMap: { [key: number]: string };
+  };
+  reportDate: string;
+  uiSchema: any;
+  defaultEmptySourceTypeState: any;
+}
+
+// 🧩 Main component
+export default function ActivityForm({
+  activityData,
+  reportDate,
+  uiSchema,
+  defaultEmptySourceTypeState,
+}: Readonly<Props>) {
+  // 🐜 To display errors
+  const [errorList, setErrorList] = useState([] as any[]);
+  // 🌀 Loading state for the Submit button
+  const [isLoading, setIsLoading] = useState(false);
+  // ✅ Success state for for the Submit button
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [formState, setFormState] = useState({} as any);
+  const [jsonSchema, setJsonSchema] = useState({});
+
+  const { activityId, sourceTypeMap } = activityData;
+
+  // Set useEffect dependency set from checked sourceTypes
+  const dependencyArray: string[] = [];
+  for (const value of Object.values(sourceTypeMap)) {
+    dependencyArray.push(
+      formState?.[`${value}`] ? formState?.[`${value}`] : null,
+    );
+  }
```

Comment:
I think we can simplify this:
```suggestion
  const dependencyArray = Object.entries(sourceTypeMap).map([k,v] => formState?.[`${v}`] ?? null)
```